cloud/cloud_env_impl.cc

```diff
@@ -1997,7 +1997,9 @@ size_t CloudEnvImpl::TEST_NumScheduledJobs() const {
 };
 
 Status CloudEnvImpl::ApplyCloudManifestDelta(const CloudManifestDelta& delta) {
-  cloud_manifest_->AddEpoch(delta.file_num, delta.epoch);
+  if (!cloud_manifest_->AddEpoch(delta.file_num, delta.epoch)) {
+    return Status::InvalidArgument("Delta already applied in cloud manifest");
+  }
```

Comment:
I don't think we should return `InvalidArgument` if the delta was already applied, given that we claim `ApplyCloudManifestDelta` is idempotent. We should return `Status::OK` and return additional information through the `bool* update_applied` argument.

InvalidArgument error code is reserved for cases where the argument is outside of the function domain and is generally indicative of an invalid state in the caller.

Reply:
Makes sense! applying old epoch is expected here while it's not expected when rolling new cookie. Will fix this in a another PR